src/rgen/Microsoft.Macios.Generator/Attributes/FieldData.cs

```diff
@@ -52,21 +56,19 @@ public static bool TryParse (AttributeData attributeData,
 			if (!attributeData.ConstructorArguments [0].TryGetIdentifier (out symbolName)) {
 				return false;
 			}
-			switch (attributeData.ConstructorArguments [1].Value) {
-			// there are two possible cases here:
-			// 1. The second argument is a string
-			// 2. The second argument is an enum
-			case T enumValue:
-				flags = enumValue;
-				break;
-			case string lib:
-				libraryName = lib;
-				break;
-			default:
-				// unexpected value :/
-				error = new (ParsingError.UnknownConstructor, attributeData.ConstructorArguments.Length);
+
+			if (attributeData.ConstructorArguments [1].Value is string) {
+				libraryName = (string?) attributeData.ConstructorArguments [1].Value!;
```

Comment:
why not:
```
if (attributeData.ConstructorArguments [1].Value is string libValue) {
    libraryName = (string?)libValue;
}
```

Reply:
I did not think about it

Reply:
now that I'm chasing it, I just found it weird to assign the variable, I wish the 'is' had a way to set the variable to be one you already have.